src/dinit-iostream.cc

```diff
@@ -0,0 +1,742 @@
+#include <limits>
+#include <cstdio>
+#include <cstring>
+
+#include <sys/types.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include <dinit-iostream.h>
+#include <baseproc-sys.h>
+
+// See dinit-iostream.h for information about the usage
+
+namespace {
+
+// A simple trick for calculating how many digits can be in a type with compile-time integer
+// logarithm like function.
+template <typename T> constexpr unsigned type_max_numdigits(T num = std::numeric_limits<T>::max(),
+        unsigned pow = 0)
+{
+    return (num == 0) ? pow : type_max_numdigits(num / 10, pow + 1);
+}
+
+constexpr unsigned UINT_MAX_CHARS = type_max_numdigits<unsigned>();
+constexpr unsigned ULONG_MAX_CHARS = type_max_numdigits<unsigned long>();
```

Comment:
Move the declarations to where they are needed.

Reply:
I didn't see this one, Sorry. I will push a fix.

(I hate GitHub when it's doesn't notify me when there is something new in the thread when you are already reading somewhere else in the thread).